shell/browser/api/electron_api_service_worker_context.cc

```diff
@@ -102,6 +103,10 @@ void ServiceWorkerContext::OnReportConsoleMessage(
            .Build());
 }
 
+void ServiceWorkerContext::OnRegistrationCompleted(const GURL& scope) {
+  Emit("registration-completed", scope);
```

Comment:
Is this just here for the test, or is it more broadly useful?

Reply:
This can be useful for tracking which extension service workers (or any others) are active. I plan to eventually make use of it for deciding which service workers to dispatch events to.

Reply:
It might be a good idea to emit an object, rather than a string, to support future extensibility, e.g. `{ scope: ... }`

Reply:
Agreed, this was done in dd5c7d69122609eff09cf682b181217a7099980b

Reply:
ah, right, sorry for the noise!